homeassistant/components/econet/entity.py

```diff
@@ -13,26 +15,26 @@ class EcoNetEntity(Entity):
 
     _attr_should_poll = False
 
-    def __init__(self, econet):
+    def __init__(self, econet: Equipment) -> None:
```

Comment:
I'd suggest to make the class generic with an upper bound and default on `Equipment`.
That would help with things like `self._econet.humidity` in `climate.py` which aren't type checked yet.